api/src/main/java/com/epam/pipeline/manager/datastorage/DataStorageManager.java

```diff
@@ -343,6 +353,40 @@ public AbstractDataStorage updatePolicy(DataStorageVO dataStorageVO) {
         return createdStorage;
     }
 
+    @Transactional(propagation = Propagation.REQUIRED)
+    public AbstractDataStorage createDefaultStorageForUser(final String userName) {
+        final String dataStorageTemplateJson =
+            preferenceManager.getPreference(SystemPreferences.DEFAULT_USER_DATA_STORAGE_TEMPLATE);
+        final DataStorageTemplate dataStorageTemplate = JsonMapper.parseData(
+            dataStorageTemplateJson.replaceAll(FolderTemplateManager.TEMPLATE_REPLACE_MARK, userName),
+            new TypeReference<DataStorageTemplate>() {});
+        final DataStorageVO dataStorageDetails = dataStorageTemplate.getDatastorage();
+        if (dataStorageDetails.getPath() == null) {
+            dataStorageDetails.setPath(adjustStoragePath(dataStorageDetails.getName(), null));
+        }
+        final AbstractDataStorage correspondingExistingStorage =
+            dataStorageDao.loadDataStorageByNameOrPath(dataStorageDetails.getName(), dataStorageDetails.getPath());
+        if (correspondingExistingStorage != null) {
+            log.warn(messageHelper.getMessage(MessageConstants.DEFAULT_STORAGE_CREATION_CORRESPONDING_EXISTS,
+                                              dataStorageDetails.getPath(),
+                                              userName,
+                                              correspondingExistingStorage.getId()));
+            return null;
+        }
+        if (!folderManager.exists(dataStorageDetails.getParentFolderId())) {
+            dataStorageTemplate.getDatastorage().setParentFolderId(null);
```

Comment:
Should we update `dataStorageDetails` here not `dataStorageTemplate`?

Reply:
Done